### PR TITLE
Fixed test mode of aws ec2_instance module

### PR DESCRIPTION
**Pull request by cahlchang**
##### SUMMARY
Fixed test mode of aws ec2_instance module

In the ec2_instance module, there was a problem to be executed regardless of the test mode.
With this PR, I will fix the problem and add the test code.

Related
https://github.com/ansible/ansible/issues/46611

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ec2_instance

##### ANSIBLE VERSION

```paste below
$ ansible --version
ansible 2.8.0.dev0 (aws_ec2_instance-fix_checkmode 7d2f8c2cb9) last updated 2018/10/11 01:14:49 (GMT +900)
  config file = None
  configured module search path = ['/home/cahlchang/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/cahlchang/git/ansible/lib/ansible
  executable location = /home/cahlchang/git/ansible/bin/ansible
  python version = 3.5.2 (default, Nov 23 2017, 16:37:01) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION

This code is testing the behavior of check mode.
https://github.com/ansible/ansible/commit/7d2f8c2cb99b1e91dd7c1478e04ce1c9fc58fe7f

In the ec2_instance module, the test was unsupported, and the integration test could not be executed.
Therefore, I ran the Playbook manually and tested.
https://github.com/ansible/ansible/blob/0c22c0cefb0c08d5bc10b72484749a5282ee220b/test/integration/targets/ec2_instance/aliases#L1-L2

* integration  test
```paste below
$ ansible-test integration ec2_instance
WARNING: Excluding tests marked "unsupported" which require --allow-unsupported or prefixing with "unsupported/": ec2_instance
WARNING: All targets skipped.

$ ansible-test sanity --test validate-modules ec2_instance
Sanity check using validate-modules
WARNING: Cannot perform module comparison against the base branch. Base branch not detected when running locally.
WARNING: Reviewing previous 1 warning(s):
WARNING: Cannot perform module comparison against the base branch. Base branch not detected when running locally.
```

* runme script

The runme script did not work due to version relation of boto3.
I think that it is necessary to fix the version of boto3.
However, I am removing it from correspondence so as to make the problem simple.

https://github.com/ansible/ansible/blob/0c22c0cefb0c08d5bc10b72484749a5282ee220b/test/integration/targets/ec2_instance/runme.sh#L19
```
+ python -m pip install 'botocore<1.10.16' boto3
Collecting botocore<1.10.16
  Using cached https://files.pythonhosted.org/packages/3e/a2/4eddf990705eccdda115d8e9e7c20fa21615704ed0806fc9eba6c2fcaad5/botocore-1.10.15-py2.py3-none-any.whl
Requirement already satisfied: boto3 in /home/cahlchang/.local/lib/python3.5/site-packages (1.7.45)
Requirement already satisfied: python-dateutil<3.0.0,>=2.1; python_version >= "2.7" in /home/cahlchang/.local/lib/python3.5/site-packages (from botocore<1.10.16) (2.7.3)
Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /home/cahlchang/.local/lib/python3.5/site-packages (from botocore<1.10.16) (0.9.3)
Requirement already satisfied: docutils>=0.10 in /home/cahlchang/.local/lib/python3.5/site-packages (from botocore<1.10.16) (0.14)
Requirement already satisfied: s3transfer<0.2.0,>=0.1.10 in /home/cahlchang/.local/lib/python3.5/site-packages (from boto3) (0.1.13)
Requirement already satisfied: six>=1.5 in /home/cahlchang/.local/lib/python3.5/site-packages (from python-dateutil<3.0.0,>=2.1; python_version >= "2.7"->botocore<1.10.16) (1.11.0)
awscli 1.15.58 has requirement botocore==1.10.57, but you'll have botocore 1.10.15 which is incompatible.
boto3 1.7.45 has requirement botocore<1.11.0,>=1.10.45, but you'll have botocore 1.10.15 which is incompatible.
```

Run manually.
```
ansible-playbook --extra-vars="@test/integration/cloud-config-aws.yml" ./test/integration/targets/ec2_instance/playbooks/full_test.yml
# test result... all passed.
```

This code also passes pep8 test.
```paste below
$ ansible-test sanity --test pep8 ec2_instance
Sanity check using pep8
```

### CODE REVIEW

**Comment by willthames** [comment]
This change looks excellent to me - thanks for doing such a great job of improving the test suite to check these changes. 

I'll try and find some time to run the test suite myself before merging this, but it has my seal of approval should anyone else be able to test this first.

**Review by willthames — APPROVED** [review]

**Review by ryansb — APPROVED** [review]

**Pull request merged**
